From 9b3135370188919c0416902d35e22b9682bc7bb0 Mon Sep 17 00:00:00 2001
From: mayi <3363523343@qq.com>
Date: Wed, 21 Oct 2026 18:34:14 +0800
Subject: [PATCH] fix(osdrv/atbm6461): match the Linux unlocked_ioctl signature

Use unsigned int for the ioctl command and the matching printf format. This is independent of SDIO transport and startup handling.
---
 extdrv/wireless/realtek/atbm6461/hal_apollo/atbm_ioctl.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/extdrv/wireless/realtek/atbm6461/hal_apollo/atbm_ioctl.c b/extdrv/wireless/realtek/atbm6461/hal_apollo/atbm_ioctl.c
index 7d72a82f010d5a8bd0e4e5964b5104bc0134e467..0052694a393e7300d561f356b7fa3e5fd909fd35 100644
--- a/extdrv/wireless/realtek/atbm6461/hal_apollo/atbm_ioctl.c
+++ b/extdrv/wireless/realtek/atbm6461/hal_apollo/atbm_ioctl.c
@@ -1974,7 +1974,7 @@ err:
 #endif
 
 #ifdef LINUX_OS
-static long atbm_unlocked_ioctl(struct file *filp, unsigned long cmd, unsigned long arg)
+static long atbm_unlocked_ioctl(struct file *filp, unsigned int cmd, unsigned long arg)
 {
 	long ret = 0;
 
@@ -2199,7 +2199,7 @@ static long atbm_unlocked_ioctl(struct file *filp, unsigned long cmd, unsigned l
 			ret = atbm_wsm_fast_cfg_send(((struct atbm_info *)(filp->private_data))->hw_priv, arg);
 			break;
 		default:
-			atbm_printk_err("%s cmd %d invalid.\n", __func__, cmd);
+			atbm_printk_err("%s cmd %u invalid.\n", __func__, cmd);
 			ret = -1;
 	}
 
